cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -221,7 +221,7 @@ func (csr *ClusterStateRegistry) updateScaleRequests(currentTime time.Time) {
 			// scale-out finished successfully
 			// remove it and reset node group backoff
 			delete(csr.scaleUpRequests, nodeGroupName)
-			csr.backoff.RemoveBackoff(scaleUpRequest.NodeGroup)
+			csr.backoff.RemoveBackoff(scaleUpRequest.NodeGroup, csr.nodeInfosForGroups[scaleUpRequest.NodeGroup.Id()])
```

Comment:
This is just sanity check - you know you may end up passing nil nodeInfo here? I think it's ok and just needs to be handled in backoff implementation. 

Also it got me thinking - what will happend if nodegroup is removed / re-configured to be no longer autoscaled during scale-up? Can this function handle scaleUp request existing for a no-longer existing NodeGroup?

Reply:
> This is just sanity check - you know you may end up passing nil nodeInfo here?

Yeah - I know that. The backoff needs to handle calls with nil NodeInfo gracefully.

> Also it got me thinking - what will happend if nodegroup is removed / re-configured to be no longer autoscaled during scale-up? Can this function handle scaleUp request existing for a no-longer existing NodeGroup?

Good point. Seems like potential bug. I will file an issue.

Reply:
#1565